backend/analysis-manager/main.py

```diff
@@ -1,5 +1,15 @@
+from typing import List
+
 import functions_framework
-from flask import Request, typing
+from flask import Request, make_response, typing
+from pydantic import BaseModel, Field, HttpUrl, ValidationError
+
+
+class AnalyzeTextAreaModel(BaseModel):
+    text_body: str = Field(min_length=10, max_length=1000)
```

Comment:
Are we sure that limiting it to 1000 characters is a good idea?

Reply:
Yes, I believe we should do this while in testing to help keep our costs down.

> The Natural Language API pricing is based on the number of Unicode characters in each request, calculated in terms of “units”. For Text Moderation, the cost per request is rounded up to the nearest unit of 100 Unicode characters and for every other feature, it's rounded to the nearest 1,000 Unicode characters. The Unicode characters include whitespace characters and any markup characters such as HTML or XML tags.
> 
> For example, if you send three requests for Sentiment Analysis that contain 800, 1,500, and 600 characters respectively, you are charged for four units: one for the first request (800), two for the second request (1,500), and one for the third request (600).

Should I leave a todo comment to change this later?

https://cloud.google.com/natural-language/pricing

Reply:
While being price-conscious is good, we also don't want to hamper our user's primary use case, which is analyzing articles. That blurb about pricing alone is ~650 characters. An article would be multiple times that.

Since we're just doing this for a project and not turning it into a business, it'll probably hurt us more than it helps us. In general, having limits is good for preventing abuse, but it always has to be balanced with the expected use cases.

If you still want to keep a limit, setting it much higher is a good idea, probably around 10,000 characters. That way, even really large articles can still be processed without issue, and it still prevents excessive abuse.

Reply:
We can discuss this further later, for now I've changed it to 5k characters.